pkg/driver/driver_openstack.go

```diff
@@ -356,16 +282,21 @@ func (d *OpenStackDriver) createNeutronClient() (*gophercloud.ServiceClient, err
 	if !ok {
 		return nil, fmt.Errorf("missing %s in secret", v1alpha1.OpenStackPassword)
 	}
+
+	// optional OS_USER_DOMAIN_NAME
+	userDomainName := d.CloudConfig.Data[v1alpha1.OpenStackUserDomainName]
+
 	domainName, ok := d.CloudConfig.Data[v1alpha1.OpenStackDomainName]
-	if !ok {
-		return nil, fmt.Errorf("missing %s in secret", v1alpha1.OpenStackDomainName)
+	domainID, ok2 := d.CloudConfig.Data[v1alpha1.OpenStackDomainID]
+	if !ok && !ok2 {
```

Comment:
So both of them have to be present right? Just confirming.

Reply:
At least one of them.